### PR TITLE
FF7: Implemented day-night time cycle

### PR DESCRIPTION
**Pull request by CosmosXIII**
## Summary

This PR implements a day-night time cycle filter that applies over background and 3d characters.
I added a new "time" folder that contains a config file to control the parameters of the day-night cycle.
The filter works both when lighting is enabled or disabled. Future improvements may be done to modify light direction with time or add weather effects.

### Motivation

I made this to make Tsuna happy. The main use case is to be used along with echo-s but it could be used with other gameplay mods. The addresses of the variables accessed from field scripts can be customized from the config file. The expected outcome is Tsuna being happy.

### ACKs

- [x] I have updated the [Changelog.md](https://github.com/julianxhokaxhiu/FFNx/blob/master/Changelog.md) file
- [x] I did test my code on FF7
- [ ] I did test my code on FF8

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -289,6 +289,10 @@ add_custom_command(
   COMMAND ${CMAKE_COMMAND} -E copy
           ${CMAKE_CURRENT_SOURCE_DIR}/misc/${RELEASE_NAME}.vibrate.ff8.battle.toml
           ${CMAKE_BINARY_DIR}/bin/vibrate/ff8/battle.toml
+  # time cycle .toml
+  COMMAND ${CMAKE_COMMAND} -E copy
+  ${CMAKE_CURRENT_SOURCE_DIR}/misc/${RELEASE_NAME}.time.toml
+  ${CMAKE_BINARY_DIR}/bin/time/config.toml
```
Comment (julianxhokaxhiu) [inline CMakeLists.txt:295]: It's minor but can you please reindent this to align with the others? :) Thanks!

**Review by julianxhokaxhiu — CHANGES_REQUESTED** [review]
LGTM, just one minor thing to be fixed and we're good to go :) Thanks!

File: CMakeLists.txt (hunk 1 above)
Comment (CosmosXIII) [reply CMakeLists.txt]: fixed!

**Review by julianxhokaxhiu — APPROVED** [review]

**Comment by julianxhokaxhiu** [comment]
Thank you!

**Pull request merged**